### DISCUSSION TITLE
Tools for Takeover

### DISCUSSION THREAD

**Discussion by b1bek** [indianajson/can-i-take-over-dns#33] [General]
Do we have any public tools to create hosted zone for services to aid us in the takeover of the vulnerable assets?

**Comment by indianajson** [comment]
@b1bek Not that I am aware of, but given the low number of targets per provider associated with VPDs and BBPs it generally isn't worth the time to handle that level of automation.

**Reply by b1bek** [reply]
So, all these takeovers done generally are manually done by most people? Asking cause never done this ever before.

**Reply by indianajson** [reply]
Most providers do not have robust APIs, thus it generally is far faster just to log in and add the domain to your account versus automating the process (since you have to manually set up the account anyway).  When it's all said and done you'd spend more time building the automation than you would save.

**Comment by 0xPrial** [comment]
@b1bek till now I have been testing all takeovers including NS Zone takeovers manually and getting great results. Just using a few loops in BASH.

**Comment by b1bek** [comment]
Created this for Azure DNS https://github.com/b1bek/NSTakeover

**Comment by Halsey1991** [comment]
Hello @indianajson &  all 

I've pulled out all my hair and I don't know where to turn for help. 
I've found vulnerability with a site I am doing pen testing for DNS takeover, the pages are vulnerable ( code 404) . However the problem is when to verify. 

I created a simple page on Shopfy and Squarespace to demonstrate the take over. However when I add the DNS it doesn't link.  Shopfy and Sqaurespace ask to delete the the other DNS entry from the other place - and says " DNS is not pointing to Shopfy" - What is the problem? Could you please help me. 

I just need to verify the DNS transfer for Hacker1 -  This has been the problem. Please see some of the error messages below. 

How do I fix this? 

![Image](https://github.com/user-attachments/assets/25dd4b4e-9f0d-4389-8458-c6f8360a9239)
![DNS problem](https://github.com/user-attachments/assets/af41ea3e-511d-4066-b5fb-e158e42ccf61)

**Reply by indianajson** [reply]
Hmm, if a page is displaying a 404 error it’s not vulnerable to the type of DNS zone takeover we cover in this repo. That’s more of a dangling DNS record issue (like Can-I-Take-Over-XYZ covers). 

But the first thing we need to do is make sure you actually control of the zone. If so, create a record and see if you can see it when you look for it on https://toolbox.googleapps.com/apps/dig/

**Reply by Halsey1991** [reply]
Hi Indi, Thanks for the reply. What do you mean take control of the zone? How do I do that? basic steps like 1,2,3... I am new to this. 
Hacker1 asks for a proof of concept and don't know how to do that. I've found several subdomains vulnerable but I don't know how to do the proof of concept - how to actually proof they can be taken over? Thanks

**Reply by indianajson** [reply]
For our type of vulnerability (DNS zone takeovers), subdomains are not usually vulnerable unless the root domain is also vulnerable. Also, if the target subdomain **displays any content whatsoever** when you visit it with a web browser then it is 100% guaranteed to **not** be vulnerable DNS zone takeover. 

What you are looking for is dangling DNS records vulnerabilities (covered by https://github.com/EdOverflow/can-i-take-over-xyz), where a subdomain is pointed to service that is misconfigured allowing you to take control of the subdomain. According to their list, neither Squarespace nor Shopify is typically vulnerable even when showing 404 errors. This is the best answer I can give you without knowing the targets.  If you're willing to provide the actual targets (you can DM them to me on X) I can explain the issue with your specific examples.

**Reply by Halsey1991** [reply]
I see, ohh I was super excited thinking I found something ...  I don't think the root domain is vulnerable. :(  but ok.  I will try to find you on X.  I thought I needed to host the page in anywhere and to add the DNS that was vulnerable.  Thank you.
